### PR TITLE
forgejo-cli: init at 0.1.1

### PR DESCRIPTION
**Pull request by isabelroses**
## Description of changes

Adds https://codeberg.org/Cyborus/forgejo-cli

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Result of `nixpkgs-review pr 333198` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>forgejo-cli</li>
  </ul>
</details>

File: pkgs/by-name/fo/forgejo-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  cargoHash = "sha256-F7UBLqMXYS8heJs1mdmiFTHUfgoMKEb+KV4tiDsIRDY=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    libgit2
+    oniguruma
+    openssl
+    zlib
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```
Comment (afh) [inline pkgs/by-name/fo/forgejo-cli/package.nix:37]: In order to successfully build on darwin the following `buildInputs` are needed:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
    Security
    SystemConfiguration
  ]);
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @isabelroses, please find below a suggestion to fix the build on darwin

**Comment by afh** [comment]
Result of `nixpkgs-review pr 333198` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>forgejo-cli</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for accepting the suggestion so swiftly, @isabelroses, much appreciated 👍

File: pkgs/by-name/fo/forgejo-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  zlib,
+  stdenv,
+  darwin,
+}:
+let
+  version = "0.1.1";
+in
+rustPlatform.buildRustPackage {
+  pname = "forgejo-cli";
+  inherit version;
```
Comment (emilylange) [inline pkgs/by-name/fo/forgejo-cli/package.nix:18]:
```suggestion

rustPlatform.buildRustPackage rec {
  pname = "forgejo-cli";
  version = "0.1.1";
```

File: pkgs/by-name/fo/forgejo-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      ]
+    );
+
+  env = {
+    RUSTONIG_SYSTEM_LIBONIG = true;
+  };
+
+  meta = {
+    description = "CLI application for interacting with Forgejo";
+    homepage = "https://codeberg.org/Cyborus/forgejo-cli";
```
Comment (emilylange) [inline pkgs/by-name/fo/forgejo-cli/package.nix:53]:
```suggestion
    homepage = "https://codeberg.org/Cyborus/forgejo-cli";
    changelog = "https://codeberg.org/Cyborus/forgejo-cli/releases/tag/${src.rev}";
```

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fo/forgejo-cli/package.nix (hunk 2 above)
Comment (emilylange) [reply pkgs/by-name/fo/forgejo-cli/package.nix:18]: Won't eval; is missing this suggestion^

Comment (emilylange) [reply pkgs/by-name/fo/forgejo-cli/package.nix:18]: Oh wait, nevermind, it does. Guess you prefer `let in` over `rec`.
Any particular reason?

Comment (isabelroses) [reply pkgs/by-name/fo/forgejo-cli/package.nix:18]: > Any particular reason?

Personal preference really. I mean I created this issue out of curiosity what others think https://github.com/NixOS/nixpkgs/issues/315337

**Review by emilylange — APPROVED** [review]
Alright then, keep your `let in` :)

**Pull request merged**